lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -458,6 +470,16 @@ export default class BitcoinProcessor {
     };
   }
 
+  /**
+   * For proof of fee calculation, blocks are grouped into fixed sized groups.
+   * This function rounds a block to the first block in its group and returns that
+   * value.
+   */
+  private roundToGroupBoundary (block: number): number {
+    const groupId = Math.floor(block / ProtocolParameters.groupSizeInBlocks);
```

Comment:
Consider renaming to `getFirstBlockInGroup`. Also consider reusing `getgroupId()`.